test/node/connection-emit-actions.js

```diff
@@ -0,0 +1,221 @@
+'use strict';
+
+const test = require('tap');
+const net = require('net');
+
+const jstp = require('../..');
+
+require('../tap-oneOf');
+
+const app = require('../fixtures/application');
+
+const application = new jstp.Application(app.name, app.interfaces);
+const serverConfig = {
+  applications: [application],
+  authPolicy: app.authCallback,
+};
+
+let server;
+let connection;
+
+test.beforeEach((done) => {
+  server = jstp.net.createServer(serverConfig);
+  server.listen(0, () => {
+    const port = server.address().port;
+    jstp.net.connect(app.name, null, port, 'localhost', (error, conn) => {
+      test.assertNot(error, 'must connect to server and perform handshake');
+      connection = conn;
+      done();
+    });
+  });
+});
+
+test.afterEach((done) => {
+  if (connection) {
+    connection.close();
+    connection = null;
+  }
+  server.close();
+  done();
+});
+
+test.test('must emit server and client events upon anonymous handshake',
+  (test) => {
+    test.plan(7);
+
+    const client = {
+      application: new jstp.Application('jstp', {}),
+    };
+
+    let serverConnection = null;
+    server.once('connect', (serverConn) => {
+      serverConnection = serverConn;
+    });
+
+    const port = server.address().port;
+    const socket = net.connect(port);
+    socket.on('error', () => {
+      test.fail('must create socket and connect to server');
+    });
+    socket.on('connect', () => {
+      serverConnection.on('handshakeRequest',
```

Comment:
I just wanted to put them in one place, also this callback should always be called after the one above, but it's okay to put them there too.